ioengines.c

```diff
@@ -466,6 +462,7 @@ enum fio_q_status td_io_queue(struct thread_data *td, struct io_u *io_u)
 		}
 	}
 
+	td->last_ddir_issued = ddir;
```

Comment:
I did not notice this in the previous version of your patch but if `ret == FIO_Q_BUSY` then the I/O was not actually issued. Assignment of `last_ddir_issued` should go in the two places where `last_was_sync` was set.

Reply:
Ah, indeed.  Let me fix this up!